mono/utils/mono-hwcap-ppc.c

```diff
@@ -62,16 +62,27 @@ mono_hwcap_arch_init (void)
 			mono_hwcap_ppc_has_multiple_ls_units = TRUE;
 	}
 #elif defined(_AIX)
-	/*
-	 * FIXME: ensure these are valid, and we match Linux ones
-	 */
-	mono_hwcap_ppc_is_isa_2x = __power_4_andup() ? TRUE : FALSE;
-	mono_hwcap_ppc_is_isa_64 = __cpu64() ? TRUE: FALSE;
+	if (__cpu64())
+		mono_hwcap_ppc_is_isa_64 = TRUE;
+	if (__power_4_andup())
+		mono_hwcap_ppc_is_isa_2x = TRUE;
+	if (__power_5_andup())
+		mono_hwcap_ppc_has_icache_snoop = TRUE;
```

Comment:
For all these consider one liners: global |= func(); instead of if (func) global = true

Reply:
The if () is also fine.